client/dhcp_client_lwt.ml

```diff
@@ -0,0 +1,90 @@
+let src = Logs.Src.create "dhcp_client_lwt"
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Make(Time : Mirage_time_lwt.S) (Net : Mirage_net_lwt.S) = struct
+  open Lwt.Infix
+
+  type lease = Dhcp_wire.pkt
+
+  type t = lease Lwt_stream.t
+
+  let connect ?(with_xid)
+              ?(requests : Dhcp_wire.option_code list option) net =
```

Comment:
I don't understand, `Net` is a `Mirage_net_lwt.S` which you'll need to have anyways (since it is actually used (`write` etc.).  What would the benefit be of passing the `Macaddr.t` explicitly instead of using the interface?  (i.e. if you have a non-network device, you'll need to implement `mac` - it is used e.g. in the arp implementation as well).

But I believe there's still enough design space to revise `Mirage_net_lwt.S` etc. when appropriate - maybe you can elaborate on your use case?